x/spec/types/combinable.go

```diff
@@ -0,0 +1,93 @@
+package types
+
+import fmt "fmt"
+
+type Combinable interface {
+	Differeniator() string
+	GetEnabled() bool
+	Equal(interface{}) bool
+}
+
+func (api *Api) Differeniator() string {
+	return api.Name
+}
+
+func (h *Header) Differeniator() string {
+	return h.Name
+}
+
+func (parsing *ParseDirective) Differeniator() string {
+	return parsing.FunctionTag.String()
+}
+
+func (e *Extension) Differeniator() string {
+	return e.Name
+}
+
+func (v *Verification) Differeniator() string {
+	return v.Name
+}
+
+// Api has GetEnabled
+
+func (h *Header) GetEnabled() bool {
+	return true
+}
+
+func (h *ParseDirective) GetEnabled() bool {
+	return true
+}
+
+func (h *Extension) GetEnabled() bool {
+	return true
+}
+
+func (h *Verification) GetEnabled() bool {
+	return true
+}
+
+func CombineFields[T Combinable](currentMap map[string]interface{}, mergedMap map[string]interface{}, combineFrom []T, toCombine []T, allowOverwrite bool) ([]T, map[string]interface{}, error) {
```

Comment:
add some comments, this method receives a lot of stuff and it is hard to understand the purpose for this